### PR TITLE
creating a new method for deleting Byon/Edge node Clouds

### PR DESCRIPTION
**Pull request by alijawadfahs**

### CODE REVIEW

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java [hunk 1]
```diff
@@ -207,49 +207,96 @@ public Boolean removeClouds(String sessionId, List<String> cloudIDs, Boolean pre
...
             try {
-                ASYNC_NODE_CANDIDATES_PROCESSES_RESULTS.add(updatingNodeCandidatesUtils.asyncClean(cloudIDs));
-            } catch (InterruptedException ie) {
-                LOGGER.warn("Thread cleaning node candidates interrupted!", ie);
+                String nodeSourceName = PACloud.WHITE_LISTED_NAME_PREFIX + cloud.getNodeSourceNamePrefix() +
+                                        entry.getKey();
+                LOGGER.info("Removing white listed IAAS node source \"{}\" from the ProActive server.", nodeSourceName);
+                resourceManagerGateway.removeNodeSource(nodeSourceName, preempt);
+            } catch (NotConnectedException | PermissionRestException | IllegalArgumentException e) {
+                LOGGER.error("Removing WL cloud crashed. Error: ", e);
```
Comment (mklkun) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java:278]: Should we not return false if one of the NSs has not been removed properly?

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java [hunk 2]
```diff
@@ -207,49 +207,96 @@ public Boolean removeClouds(String sessionId, List<String> cloudIDs, Boolean pre
...
+    public Boolean removeByonCloudNS(String sessionId, PACloud cloud, Boolean preempt) {
+        LOGGER.info("Removing {} cloud : {}", cloud.getCloudType(), cloud.toString());
+        try {
+            String nodeSourceName = cloud.getCloudID();
+            LOGGER.info("Removing {} node source \"{}\" from the ProActive server.",
+                        cloud.getCloudType(),
+                        nodeSourceName);
+            resourceManagerGateway.removeNodeSource(nodeSourceName, preempt);
+        } catch (NotConnectedException | PermissionRestException | IllegalArgumentException e) {
+            LOGGER.error("Removing cloud crashed. Error: ", e);
```
Comment (mklkun) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java:301]: Same here.

**Review by mklkun — COMMENTED** [review]
Some comments regarding the behaviour

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java (hunk 1 above)
Comment (alijawadfahs) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java:278]: True implementing a flag to handle this issue

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java (hunk 2 above)
Comment (alijawadfahs) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java:301]: same here

**Review by mklkun — APPROVED** [review]
LGTM

**Pull request merged**